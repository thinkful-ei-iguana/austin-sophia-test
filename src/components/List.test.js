import React from 'react';
import List from './List';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    const props = {
        id: '3',
        header: 'Third list',
        cardIds: [ 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm' ],
    };

    ReactDOM.render(<List list={props} />, div);
    ReactDOM.unmountComponentAtNode(div);
}); 

it('renders the UI as expected', () =>{

    const tree = renderer
      .create(<section class="List"><header class="List-header">First list</header><div class="List-cards"><div class="Card"><h3>First card</h3><p>lorem ipsum</p><button>Delete</button></div><div class="Card"><h3>Second card</h3><p>lorem ipsum</p><button>Delete</button></div><div class="Card"><h3>Fifth card</h3><p>lorem ipsum</p><button>Delete</button></div><div class="Card"><h3>Sixth card</h3><p>lorem ipsum</p><button>Delete</button></div><div class="Card"><h3>Seventh card</h3><p>lorem ipsum</p><button>Delete</button></div><div class="Card"><h3>Tenth card</h3><p>lorem ipsum</p><button>Delete</button></div><div class="Card"><h3>Twelfth card</h3><p>lorem ipsum</p><button>Delete</button></div><div class="Card"><h3>Thirteenth card</h3><p>lorem ipsum</p><button>Delete</button></div></div><button class="List-add-button">+ Random Card</button></section>)
      .toJSON();
    expect(tree).toMatchSnapshot();
});
