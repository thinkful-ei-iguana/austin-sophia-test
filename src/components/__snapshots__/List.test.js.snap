// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<section
  class="List"
>
  <header
    class="List-header"
  >
    First list
  </header>
  <div
    class="List-cards"
  >
    <div
      class="Card"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
      <button>
        Delete
      </button>
    </div>
    <div
      class="Card"
    >
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
      <button>
        Delete
      </button>
    </div>
    <div
      class="Card"
    >
      <h3>
        Fifth card
      </h3>
      <p>
        lorem ipsum
      </p>
      <button>
        Delete
      </button>
    </div>
    <div
      class="Card"
    >
      <h3>
        Sixth card
      </h3>
      <p>
        lorem ipsum
      </p>
      <button>
        Delete
      </button>
    </div>
    <div
      class="Card"
    >
      <h3>
        Seventh card
      </h3>
      <p>
        lorem ipsum
      </p>
      <button>
        Delete
      </button>
    </div>
    <div
      class="Card"
    >
      <h3>
        Tenth card
      </h3>
      <p>
        lorem ipsum
      </p>
      <button>
        Delete
      </button>
    </div>
    <div
      class="Card"
    >
      <h3>
        Twelfth card
      </h3>
      <p>
        lorem ipsum
      </p>
      <button>
        Delete
      </button>
    </div>
    <div
      class="Card"
    >
      <h3>
        Thirteenth card
      </h3>
      <p>
        lorem ipsum
      </p>
      <button>
        Delete
      </button>
    </div>
  </div>
  <button
    class="List-add-button"
  >
    + Random Card
  </button>
</section>
`;
